ImageInLib/ImageInLib/src/imageInterpolation.c

```diff
@@ -3,255 +3,200 @@
-            for (j = 0; j < width; j++) {
+    int i_int = 0, j_int = 0;
 
-                indx = x_new(i, j, length);
-                in = 2 * i; jn = j * 2; kn = k * 2;
+    double distP11 = getPoint2DDistance(point, P11);
+    if (distP11 < min_dist) {
+        min_dist = distP11;
+        i_int = i_floor;
+        j_int = j_floor;
+    }
 
-                newImage[kn][x_new(in, jn, lengthNew)] = originalImage[k][indx];
-                newImage[kn][x_new(in + 1, jn, lengthNew)] = originalImage[k][indx];
-                newImage[kn][x_new(in, jn + 1, lengthNew)] = originalImage[k][indx];
-                newImage[kn][x_new(in + 1, jn + 1, lengthNew)] = originalImage[k][indx];
+    double distP12 = getPoint2DDistance(point, P12);
+    if (distP12 < min_dist) {
+        min_dist = distP12;
+        i_int = i_floor;
+        j_int = j_ceil;
+    }
 
-                newImage[kn + 1][x_new(in, jn, lengthNew)] = originalImage[k][indx];
-                newImage[kn + 1][x_new(in + 1, jn, lengthNew)] = originalImage[k][indx];
-                newImage[kn + 1][x_new(in, jn + 1, lengthNew)] = originalImage[k][indx];
-                newImage[kn + 1][x_new(in + 1, jn + 1, lengthNew)] = originalImage[k][indx];
-            }
-        }
+    double distP21 = getPoint2DDistance(point, P21);
+    if (distP21 < min_dist) {
+        min_dist = distP21;
+        i_int = i_ceil;
+        j_int = j_floor;
     }
 
-    return true;
-}
+    double distP22 = getPoint2DDistance(point, P22);
+    if (distP22 < min_dist) {
+        min_dist = distP22;
+        i_int = i_ceil;
+        j_int = j_ceil;
+    }
 
-/*
- * Get 3D point in real world from 3D image point
- * Three operations are done here : image scaling, image translation and image rotation
-*/
-Point3D imageCoordToRealCoord(Point3D srcPoint, Point3D imageOrigin, VoxelSpacing imageSpacing, OrientationMatrix orientation) {
-    Point3D resultPoint;
-    resultPoint.x = imageOrigin.x + imageSpacing.sx * (orientation.v1.x + orientation.v1.y + orientation.v1.z) * srcPoint.x;
-    resultPoint.y = imageOrigin.y + imageSpacing.sy * (orientation.v2.x + orientation.v2.y + orientation.v2.z) * srcPoint.y;
-    resultPoint.z = imageOrigin.z + imageSpacing.sz * (orientation.v3.x + orientation.v3.y + orientation.v3.z) * srcPoint.z;
-    return resultPoint;
+    //set the interpolated value
+    return src_image.imageDataPtr[x_new(i_int, j_int, src_image.height)];
 }
 
+dataType getInterpolatedValueBilinear2D(Image_Data2D src_image, Point2D point) {
 
-//Get 3D point in image coordinates system from 3D point in real coordinate system
-//TO DO : find manually the matrix inverse in order to write the final function 
-Point3D realCoordToImageCoord(Point3D srcPoint, Point3D realOrigin, VoxelSpacing imageSpacing, OrientationMatrix orientation) {
-    Point3D resultPoint;
+    int i_floor, i_ceil, j_floor, j_ceil;
 
-    return resultPoint;
-}
+    if (floor(point.x / src_image.spacing.sx) > src_image.height - 1) {
```

Comment:
Here we are in different case, compared to nearest neighbor. We are not looking for closest point, but we consider all neighboring points always, so src_image.spacing should not be used here.